### PR TITLE
Fix JSON float formatting

### PR DESCRIPTION
**Pull request by pySilver**
See #1634 for details

### CODE REVIEW

**Comment by ruflin** [comment]
Interesting addition, didn't even know this constant exists. I'm trying to think of any existing deployments we might break with this. In the case someone defined it as `integer` on the Elasticsearch side already and a float is sent, as far as I know it will just convert it back to integer and keep working. Any ideas on what it could break?

@ezimuel Any thoughts related to this?

@pySilver Could you also add a CHANGELOG entry and some tests to it?

**Comment by pySilver** [comment]
@ruflin I can't see any way it can brake a valid setup here since on both sides data will be in a proper/expected type.

Btw, its a default JSON serialization flag in elasticsearch-php library for a long time already: https://github.com/elastic/elasticsearch-php/pull/481

I'll add changelog entry in a moment, but I'm unable to add tests (I've switched back from python not so long ago and not yet familiar with PHP tests yet)

**Comment by ezimuel** [comment]
@ruflin, I agree to add `JSON_PRESERVE_ZERO_FRACTION` for json_encode, we used it in [elasticserach-php](https://github.com/elastic/elasticsearch-php/pull/481), as suggested by @pySilver.
I'm not sure about the fix proposed by @pySilver, since the `JSON::stringify` is also used in other Elastica classes. Instead of change all the stringify calls why don't add  `JSON_PRESERVE_ZERO_FRACTION` directly in `JSON::stringify` as default parameter?

**Comment by ruflin** [comment]
Wasn't aware we also use it already in the official client. ++ on adding it then :-)

@pySilver WDYT about the proposal from @ezimuel ?

**Comment by pySilver** [comment]
@ruflin I think its totally correct and should be a default behavior for all serialization tasks.

**Comment by ruflin** [comment]
@pySilver Could you modify your PR to this approach?

**Comment by pySilver** [comment]
@ruflin done!

**Comment by ruflin** [comment]
Thanks. Could you try to run `make lint`. For some reason the linter in CI is happy. The strange thing is the file it complains about is not touched in this PR?!?

**Comment by pySilver** [comment]
@ruflin yeah, same thing on localhost. The reason is here: https://github.com/ruflin/Elastica/blob/master/lib/Elastica/Index/Settings.php#L207 introduced 4m ago at https://github.com/ruflin/Elastica/pull/1569

(Bool should be bool), changing it in my pr.

File: lib/Elastica/JSON.php [hunk 1]
```diff
@@ -60,6 +60,9 @@ public static function stringify($args/* inherit from json_encode */)
         // extract arguments
         $args = \func_get_args();
 
+        // set defaults
+        isset($args[1]) ? $args[1] |= JSON_PRESERVE_ZERO_FRACTION : $args[1] = JSON_PRESERVE_ZERO_FRACTION;
```
Comment (ruflin) [inline lib/Elastica/JSON.php]: Had to read this line a few times but makes sense now :-)

**Comment by ruflin** [comment]
@pySilver Thanks for the fix, not sure how that Bool thingy slipped through for so long.

File: lib/Elastica/JSON.php (hunk 1 above)
Comment (pySilver) [reply lib/Elastica/JSON.php]: Yeah, kinda cryptic. I couldn't find easier way to append const if arg[1] is set/not set.

Comment (chx) [reply lib/Elastica/JSON.php]: `$args[1] = ($args[1] ?? 0) | JSON_PRESERVE_ZERO_FRACTION;`

Comment (ruflin) [reply lib/Elastica/JSON.php]: Thanks for chiming in @chx . Should we update the code?

Comment (chx) [reply lib/Elastica/JSON.php]: I think so :)

Comment (ruflin) [reply lib/Elastica/JSON.php]: @chx @pySilver Someone wants to take it? ;-)

**Pull request merged**